google/cloud/compute/samples/disk_samples.cc

```diff
@@ -0,0 +1,138 @@
+  using ::google::cloud::make_ready_future;
+  using ::google::cloud::StatusOr;
+
+  [](std::string const& project_id, std::string const& zone,
+     std::string const& disk_name, std::string const& disk_size_gb,
+     std::vector<std::pair<std::string, std::string>> const& labels) {
+    auto client =
+        compute_disks::DisksClient(compute_disks::MakeDisksConnectionRest());
+    compute_proto::Disk disk;
+    disk.set_name(disk_name);
+    disk.set_size_gb(disk_size_gb);
+    for (auto const& l : labels) {
+      (*disk.mutable_labels())[l.first] = l.second;
+    }
+    StatusOr<compute_proto::Disk> result =
+        client.InsertDisk(project_id, zone, disk)
+            .then([client, project_id, zone, disk_name](auto f) mutable {
+              StatusOr<compute_proto::Operation> create_result = f.get();
+              if (!create_result) {
+                return make_ready_future<StatusOr<compute_proto::Disk>>(
+                    std::move(create_result).status());
+              }
+              auto get_disk = client.GetDisk(project_id, zone, disk_name);
+              return make_ready_future(get_disk);
+            })
+            .get();
+    if (!result) throw std::move(result).status();
+    std::cout << "Created disk: " << result->DebugString() << "\n";
+  }
+  //! [END compute_disk_create_empty_disk]
+  (argv.at(0), argv.at(1), argv.at(2), argv.at(3),
+   SplitLabelsString(argv.at(4)));
+}
+
+void DeleteDisk(std::vector<std::string> const& argv) {
+  if (argv.size() != 3) {
+    throw google::cloud::testing_util::Usage{
+        "compute-disk-delete-disk <project-id> <zone> <disk-name>"};
+  }
+  //! [START compute_disk_delete]
+  namespace compute_disks = ::google::cloud::compute_disks_v1;
+  [](std::string const& project_id, std::string const& zone,
+     std::string const& disk_name) {
+    auto client =
+        compute_disks::DisksClient(compute_disks::MakeDisksConnectionRest());
+    auto delete_result = client.DeleteDisk(project_id, zone, disk_name).get();
+    if (!delete_result) throw std::move(delete_result).status();
+    std::cout << "Deleted disk: " << disk_name << "\n";
+  }
+  //! [END compute_disk_delete]
+  (argv.at(0), argv.at(1), argv.at(2));
+}
+
+void AutoRun(std::vector<std::string> const& argv) {
+  using ::google::cloud::internal::GetEnv;
+  namespace examples = ::google::cloud::testing_util;
+
+  if (!argv.empty()) throw examples::Usage{"auto"};
+  examples::CheckEnvironmentVariablesAreSet(
+      {"GOOGLE_CLOUD_PROJECT", "GOOGLE_CLOUD_CPP_TEST_ZONE"});
+  auto const project_id = GetEnv("GOOGLE_CLOUD_PROJECT").value();
+  auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  auto const zone = GetEnv("GOOGLE_CLOUD_CPP_TEST_ZONE").value();
+  std::string disk_name = [&] {
+    return absl::StrCat(
+        "int-test-disk-",
+        google::cloud::internal::Sample(
+            generator, 8, "abcdefghijklmnopqrstuvwxyz0123456789"));
```

Comment:
optional/comment: we often encode the date into these names, so we can clean up stale resources...

But I am in favor of waiting until we notice any leaks to do so.